source/base/Shmoose.cpp

```diff
@@ -123,7 +123,30 @@ void Shmoose::mainConnect(const QString &jid, const QString &pass)
 {
     persistence_->openDatabaseForJid(jid);
 
-    QString completeJid = jid + "/shmoose";
+    QString resourceName;
+    QString resourceId;
+
+    //generate unique resource name
+    resourceId = settings_->getResourceId();
+    if(resourceId.isEmpty())
+    {
+       const QString possibleCharacters("ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz");
+       const int randomStringLength = 4; 
+
+       QString randomString;
+       for(int i=0; i<randomStringLength; ++i)
+       {
+           int index = qrand() % possibleCharacters.length();
+           QChar nextChar = possibleCharacters.at(index);
+           randomString.append(nextChar);
+       }
+
+       resourceId = randomString;
+       settings_->setResourceId(resourceId);
+    }
+
```

Comment:
This could also produce the same id on different devices. What about using this:

https://sailfishos.org/develop/docs/sailfish-mdm/sailfish-mdm-sysinfo.html/
QString SysInfo::deviceUid()

or a solution, which does also work on a desktop build:

QtSystemInfo::DeviceInfo::uniqueDeviceID() 

With this, the need of generating a random is eliminated and also the need of storing this random is gone.

Reply:
ok. I will do it this way

Reply:
Hi, I used this API but there is an issue when installing the application. A pop-up shows up claiming that Remote Management (mdm) has been activated on the device and I should contact a sysadmin if I want to know more. This is a little bit scaring. I think this is not a good idea to rely on it. I suggest to rollback to my initial proposal i.e. to use a random string and to store it in the config file. Do you agree?

Reply:
Indeed, that sounds strange and it should be avoided. Hopefully QtSystemInfo::DeviceInfo::uniqueDeviceID() works better.

Reply:
I didn't find QtSystemInfo::DeviceInfo::uniqueDeviceID()

Reply:
Just checked it. Yes, this is a qml function. Not really suitable for your need. Sorry. There are new qt classes which provide uniqueId functions, but not in that version Sailfish supports :-(. With that having said, I would vote to use your implementation. Just one more thing. Can you move the resourceId generartion into the System class as a static function? Something like System::generateResourceId()? Shmoose.cpp is already blown up, that class should not have more lines that really necessary.

Reply:
Yes.  No pb I'm going to do it this way.